src/cpu/aarch64/acl_convolution_utils.cpp

```diff
@@ -247,76 +281,38 @@ status_t init_conf_indirect_gemm(acl_conv_conf_t &acp, memory_desc_t &src_md,
         memory_desc_t &weights_md, memory_desc_t &dst_md,
         memory_desc_t &bias_md, const convolution_desc_t &cd,
         const primitive_attr_t &attr) {
-    // Indirect convolution results in slowdown for low thread count or 1x1
-    // kernels, so fall back to GEMM-based convolution in these cases
-    if (one_of(true, weights_md.dims[2] == 1, // kh
-                weights_md.dims[3] == 1, // kw
-                dnnl_get_max_threads() < 28)) {
+
+    if (weights_md.dims[2] == 1 && weights_md.dims[3] == 1)
         return status::unimplemented;
-    }
 
     CHECK(acl_init_conf(acp, src_md, weights_md, dst_md, bias_md, cd, attr));
 
+    // Indirect is slower than gemm for low thread counts, except for fast math
+    // where gemm conv is not supported yet
+    if (dnnl_get_max_threads() < 28 && !acp.fast_math)
+        return status::unimplemented;
+
+    // If we do not need to pad input channels for fast math mode then it would
+    // be faster to run convolution with im2row instead of using indirect kernel
+    int block_by = arm_compute::block_by(acp.weights_info.weight_format());
+    int ic = src_md.dims[1];
+    if (acp.fast_math && ic % block_by == 0) return status::unimplemented;
+
+    // TODO: remove this once NEGEMMConv2d::validate allows src and weights to mismatch
+    acp.wei_tensor_info.set_data_layout(arm_compute::DataLayout::NHWC);
+
     // clang-format off
     // NOTE: indirect convolution method supports only nhwc layout.
     ACL_CHECK_VALID(arm_compute::NEGEMMConv2d::validate(
-        &acp.src_info,
-        &acp.wei_info,
-        acp.with_bias ? &acp.bia_info : nullptr,
-        &acp.dst_info,
+        &acp.src_tensor_info,
+        &acp.wei_tensor_info,
+        acp.with_bias ? &acp.bia_tensor_info : nullptr,
+        &acp.dst_tensor_info,
         arm_compute::Conv2dInfo(acp.padstride_info,
                                 acp.dilation_info,
                                 acp.act_info,
                                 acp.fast_math,
-                                1)));
-    // clang-format on
-
-    return status::success;
-}
-
-status_t init_conf_wino(acl_conv_conf_t &acp, memory_desc_t &src_md,
```

Comment:
Did you remove Winograd implementation completely? If so, please move it to a separate commit.